app/controllers/admin/api/buyers_applications_controller.rb

```diff
@@ -46,7 +48,8 @@ def create
     application = applications.new(user_account: buyer, plan: application_plan, create_origin: "api")
     application.unflattened_attributes = application_params
     application.user_key = params[:user_key] if params[:user_key]
-    application.application_id = params[:application_id] if params[:application_id]
+    application_id_param = params[:application_id]
```

Comment:
Why don't we use `require` here and [here](https://github.com/3scale/porta/pull/2771/files#diff-6d426f7a2ef18db9c2765aac3763814c7c2a8eca35c260faa1e98f31523a3f29R101)?
```suggestion
    application_id_param = params.require(:application_id)
```

Reply:
@josemigallas it would break the logic. `application_id` is not a required parameter. (Application can also have only user_key without  app_id)

You can also see it from:

```
 ##~ op.parameters.add :name => "application_id", :description => "App ID or Client ID (for OAuth and OpenID Connect authentication modes) of the application to be created.", :dataType => "string", :allowMultiple => false, :required => false, :paramType => "query"
```